ssr/render.js

```diff
@@ -214,12 +200,15 @@ export default function render(
   }
 
   if (possibleLocales) {
-    const possibleLocalesTag = `<script>window.__possibleLocales__ = JSON.parse(${serializeDocumentData(
-      possibleLocales
-    )});</script>`;
-    $("#root").after(possibleLocalesTag);
+    hydrationData.possibleLocales = possibleLocales;
   }
 
+  $("#root").after(
+    `<script type="application/json" id="hydration">${JSON.stringify(
+      hydrationData
+    )}</script>`
+  );
```

Comment:
I vaguely remembered you digging into that but couldn't remember that blog anymore. Thank you for relinking it!

I started playing around with my own benchmark, which I imagine has some short-comings that I have not considered yet but I'll already leave it here to keep the conversation going. So while I did get the opposite results of the blog, I don't trust my benchmark yet. Will take a walk now and figure out how to make it better later. Anyway here's the (probably very wrong) bench and results:
https://gist.github.com/Gregoor/52a12200f8be688f7168c7dfcff422c1

JSON:
<img width="349" alt="Screenshot 2021-04-28 at 12 08 26" src="https://user-images.githubusercontent.com/4051932/116386827-7033ef80-a81a-11eb-88d6-d51507291fa2.png">

JSESC:
<img width="349" alt="Screenshot 2021-04-28 at 12 08 47" src="https://user-images.githubusercontent.com/4051932/116386862-7d50de80-a81a-11eb-893d-449a4a566472.png">

Reply:
okay thought about it some more, I guess that makes sense because Lighthouse's TTI does not mean that the app is responding yet. Since the JSON is not parsed at this point, the button click (which depends on the hydration data being loaded and parsed) does not do anything yet.
So TL;DR: I need to create a bench that is less naive about what TTI means

Reply:
web.dev [does say](https://web.dev/interactive/#what-tti-measures)
> TTI measures how long it takes a page to become fully interactive. A page is considered fully interactive when:
> * ...
> * Event handlers are registered for most visible page elements, and

but from the numbers it looks like it might not be doing that correctly. I will try to rerun it later without the event listeners attached which should make the numbers smaller, and if it doesn't, then my suspicion is confirmed.

Reply:
We chatted about this at length on Slack and I think can close it. 
I wish we had a URL to point to in a code comment that records the decision made here. Because, perhaps the next MDN dev comes along in a year and also thinks back to that original tweet by Martin B and the blog post by Henrik J.

I tested your benchmark but with "realistic" JSON from Yari. It's still not really realistic because I dug up the largest `index.json` I could find. 
But what I found was that it was virtually no difference. If anything, I got a 0.1s improvement with the `json` variant on "Time to interactive" and "Largest Contentful Paint". 
The other thing I noticed was that I couldn't even find the JSON parsing or the JS parsing in the Chrome Performance dev tools. It's there but it's so tiny (even at 6x CPU throttle) that it barely shows up. 

The advantages are nevertheless overwhelming in favor of avoiding a `<script type="text/javascript">` so we *can* have a CSP! 

I'd press the "Resolve conversion" button but then this juicy discussion would disappear.